TerriaJS/terriajs
# Add EPSG:7844 to fix ideal zoom for esri mapserver item.

### What this PR does

Fixes #6590 

- Adds proj4 definition for EPSG:7844 to Terria
  This is required to fix #6590.
- Converted modules `Proj4Defintions` and `Reproject` to Typescript.
- Enable spec for Reproject module.

### Test me

- Open this CI link for [main branch](http://ci.terria.io/main/#share=s-dpFrV1KOQnCzf8KVQBXQeSlfdu)
- Observe that the `Ideal zoom` workbench feature doesn't work for any of the dataset.
- Now open CI for [this branch](http://ci.terria.io/add-crs-7844/#share=s-dpFrV1KOQnCzf8KVQBXQeSlfdu).
- Observe that the `Ideal zoom` is now fixed.

### Checklist
